lib/pipeline.ts

```diff
@@ -330,7 +333,13 @@ Pipeline.prototype.exec = function (callback: CallbackFunction) {
   if (this.isCluster) {
     return pMap(scripts, (script) => _this.redis.script("load", script.lua), {
       concurrency: 10,
-    }).then(execPipeline);
+    })
+      .then(function () {
+        for (let i = 0; i < scripts.length; i++) {
+          _this.redis._addedScriptHashes[scripts[i].sha] = true;
+        }
+      })
```

Comment:
This promise is not strictly needed so I would just do a `return execPipeline()` at its end.